firstvoices/backend/views/mtd_data_views.py

```diff
@@ -47,17 +54,37 @@ class MTDSitesDataViewSet(
 ):
     http_method_names = ["get"]
     renderer_classes = [JSONRenderer]  # no camel-case for this data format
+    permission_type_map = {
+        **FVPermissionViewSetMixin.permission_type_map,
+        "task": None,
+    }
 
     def list(self, request, *args, **kwargs):
         site = self.get_validated_site()
-        mtd_exports_for_site = (
-            MTDExportFormat.objects.filter(site=site)
-            .filter(is_preview=False)
-            .only("latest_export_result")
+        mtd_exports_for_site = MTDExportFormat.objects.filter(
+            site=site, status=JobStatus.COMPLETE
+        ).only("export_result")
+
+        if mtd_exports_for_site:
+            return Response(mtd_exports_for_site.latest().export_result)
+        return Response(
+            {
+                "message": "Site has not been successfully indexed yet. MTD export format not found."
+            },
+            status=status.HTTP_404_NOT_FOUND,
         )
 
+    @action(detail=False, methods=["get"])
+    def task(self, request, *args, **kwargs):
+        site = self.get_validated_site()
+        mtd_exports_for_site = filter_by_viewable(
+            request.user, MTDExportFormat.objects.filter(site=site)
+        )
         if mtd_exports_for_site:
-            return Response(mtd_exports_for_site.latest().latest_export_result)
-        return HttpResponseNotFound(
-            "Site has not been indexed yet. MTD export format not found."
+            return Response(
+                MTDExportFormatTaskSerializer(mtd_exports_for_site.latest()).data
```

Comment:
This makes manually confirming that an export_job was cancelled or that an export_job failed difficult as the job instances are deleted after a successful job completes. This shouldn't be a huge issue but if we'd rather keep a log/list of jobs that have been completed I am open to suggestions.

Reply:
If there is a possibility that we might need to refer back to those records, +1 for keeping a log/list for all the jobs that completed or failed, without storing there export-data possibly ?
But since we can rebuild the mtd-Index as required, and it is not dependent on the previous export_job, I believe we should be good with cleaning out old jobs for now.

Reply:
To remedy this I have made the mtd-data/tasks/ endpoint able to serialize multiple models, and multiple models will only exist if there have been multiple failed/cancelled jobs. This means we can see all consecutive mtd export jobs that completed with problems, or just the latest job if the build was successful.